src/fileformats/ocd_types.h

```diff
@@ -308,6 +308,27 @@ namespace Ocd
 		FramingRectangle = 3  /// \since V8; Not for line text symbols
 	};
 	
+	/**
+	 * Returns pointer to a structure in a QByteArray where the referenced
+	 * structure is guaranteed to be within the bounds of the array.
+	 *
+	 * Nullptr is returned if part of the structure would be outside of
+	 * the array. Caller if expected to check for this condition.
+	 */
+	template< typename StructureType >
+	StructureType* getPointerChecked(const QByteArray& byte_array,
+	                                 quint32 offset)
+	{
+		if (Q_UNLIKELY(offset + sizeof(StructureType) > byte_array.size()))
+		{
+			qWarning("Attempt to access misplaced file structure (%s@%d)",
+			         typeid(StructureType).name(), offset);
+			return nullptr;
+		}
+
+		return reinterpret_cast< StructureType* >
+		        (const_cast< char* >(byte_array.data()) + offset);
+	}
```

Comment:
This is almost identical with what is already implemented by `Ocd::getBlockChecked`.

(`Ocd::getBlockChecked` returns `nullptr` also when `offset` is zero.)